src/encoded/static/components/navigation/components/SearchBar.js

```diff
@@ -28,59 +29,121 @@ export class SearchBar extends React.PureComponent{
         super(props);
         this.onToggleVisibility     = this.onToggleVisibility.bind(this);
-        this.toggleSearchAllItems   = this.toggleSearchAllItems.bind(this);
+        this.onChangeSearchItemType = this.onChangeSearchItemType.bind(this);
         this.onSearchInputChange    = this.onSearchInputChange.bind(this);
         this.onResetSearch          = this.onResetSearch.bind(this);
         this.onSearchInputBlur      = this.onSearchInputBlur.bind(this);
-        this.selectItemTypeDropdown = this.selectItemTypeDropdown.bind(this);
+        this.getDropdownTitleText   = this.getDropdownTitleText.bind(this);
+        this.getActionType          = this.getActionType.bind(this);
 
         let initialQuery = '';
         if (props.href){
             initialQuery = searchFilters.searchQueryStringFromHref(props.href) || '';
         }
 
+        this.memoized = {
+            deriveSearchItemTypeFromHref: memoize(SearchBar.deriveSearchItemTypeFromHref),
+            buildURIQuery: memoize(SearchBar.buildURIQuery)
+        };
+
         this.state = {
-            'searchAllItems'    : props.href && navigate.isSearchHref(props.href),
+            'searchItemType'    : this.memoized.deriveSearchItemTypeFromHref(props.href, null),
             'typedSearchQuery'  : initialQuery,
             'isVisible'         : isSelectAction(props.currentAction) || SearchBar.hasInput(initialQuery) || false
         };
 
         this.inputElemRef = React.createRef();
     }
 
-    componentDidUpdate(pastProps){
+    componentDidUpdate(pastProps) {
         const { href, currentAction } = this.props;
         const { href: pastHref } = pastProps;
 
-        if (pastHref !== href){
-            // Since is PureComponent, if state values are updated to be same value, will not update/rerender.
-            this.setState(function(currState){
+        if (pastHref !== href) {
+            this.setState(({ isVisible, searchItemType })=>{
                 const typedSearchQuery = searchFilters.searchQueryStringFromHref(href) || '';
                 return {
                     // We don't want to hide if was already open.
-                    isVisible : isSelectAction(currentAction) || currState.isVisible || SearchBar.hasInput(typedSearchQuery) || false,
-                    typedSearchQuery
+                    isVisible : isSelectAction(currentAction) || isVisible || SearchBar.hasInput(typedSearchQuery) || false,
+                    typedSearchQuery,
+                    searchItemType: this.memoized.deriveSearchItemTypeFromHref(href, searchItemType)
                 };
             });
         }
     }
 
-    toggleSearchAllItems(willSearchAllItems = !this.state.searchAllItems){
-        this.setState(function({ searchAllItems }){
-            if (typeof willSearchAllItems === 'boolean'){
-                if (willSearchAllItems === searchAllItems){
-                    return null;
+    getDropdownTitleText() {
+        const { href } = this.props;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['href' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4513406)